app/controllers/concerns/shopify_app/shop_access_scopes_verification.rb

```diff
@@ -13,11 +13,7 @@ module ShopAccessScopesVerification
 
     def login_on_scope_changes
       if scopes_mismatch?
-        if embedded_param?
-          redirect_for_embedded
-        else
-          redirect_to(shop_login)
-        end
+        fullpage_redirect_to(reauthorized_shop_login_url)
```

Comment:
I'm still considering if this would be a suitable approach for all apps.

Reply:
I think it is, but it relies on the "old" CDN-based app bridge to do the actual redirecting.